### PR TITLE
19: Redux V - Redux Recompose

### PR DESCRIPTION
**Pull request by gabrielizalo**
# Summary

* Now Gabo's Tic Tac Toe app is using Redux Recompose.

# Trello Card

https://trello.com/c/MWmkf0nq/17-redux-v-redux-recompose-todo-finger

# Screenshot

gabo s tic tac toe 2018-11-27 15-43-09

### CODE REVIEW

File: react/react-training/src/redux/login/actions.js [hunk 1]
```diff
@@ -1,55 +1,31 @@
...
-  // Login Service
-  USER_LOGIN: 'USER_LOGIN',
-  USER_LOGIN_FAILURE: 'USER_LOGIN_FAILURE',
-  USER_LOGIN_SUCCESS: 'USER_LOGIN_SUCCESS',
-
-  // Other actions
-  USER_LOGIN_VERIFY: 'USER_LOGIN_VERIFY',
-  USER_LOGOUT: 'USER_LOGOUT'
-};
+const myActionTypes = ['USER_LOGIN'];
```
Comment (henryzarza) [inline react/react-training/src/redux/login/actions.js]: I think that isn't necessary this variable, you can assign directly in the `completeTypes`

File: react/react-training/src/redux/login/actions.js [hunk 2]
```diff
@@ -1,55 +1,31 @@
...
-        payload: {
-          userIsLogged,
-          userLoginError
-        }
-      });
-    }
-  },
+  userLogin: (username, password) => ({
+    type: actionTypes.USER_LOGIN,
+    target: 'userLogin',
```
Comment (henryzarza) [inline react/react-training/src/redux/login/actions.js]: To put this value in a constant

File: react/react-training/src/redux/login/actions.js [hunk 3]
```diff
@@ -1,55 +1,31 @@
...
+    payload: { username, password },
+    injections: [
+      withPostSuccess((dispatch, response) => {
+        const userIsLogged = true;
+        const userSession = response.data.id;
+        localStorage.setItem('userIsLogged', `${userIsLogged}`);
+        localStorage.setItem('userSession', userSession);
+        dispatch({
+          type: actionTypes.USER_LOGIN_SET,
+          target: 'userIsLogged',
```
Comment (henryzarza) [inline react/react-training/src/redux/login/actions.js]: Move this to constant

File: react/react-training/src/redux/login/actions.js [hunk 4]
```diff
@@ -1,55 +1,31 @@
...
-    }
-  },
+  userLogin: (username, password) => ({
+    type: actionTypes.USER_LOGIN,
+    target: 'userLogin',
+    service: UserService.userLogin,
+    payload: { username, password },
+    injections: [
+      withPostSuccess((dispatch, response) => {
+        const userIsLogged = true;
```
Comment (henryzarza) [inline react/react-training/src/redux/login/actions.js]: I think that this variable isn't necessary, you can assign this value directly

File: react/react-training/src/redux/login/actions.js [hunk 5]
```diff
@@ -1,55 +1,31 @@
...
+        const userSession = response.data.id;
```
Comment (henryzarza) [inline react/react-training/src/redux/login/actions.js]: The same here

File: react/react-training/src/redux/user/actions.js [hunk 6]
```diff
@@ -1,109 +1,28 @@
...
-  USER_PATCH: 'USER_PATCH',
-  USER_PATCH_FAILURE: 'USER_PATCH_FAILURE',
-  USER_PATCH_SUCCESS: 'USER_PATCH_SUCCESS',
-
-  // Post User Service
-  USER_POST: 'USER_GET',
-  USER_POST_FAILURE: 'USER_GET_FAILURE',
-  USER_POST_SUCCESS: 'USER_GET_SUCCESS'
-};
+const myActionTypes = ['USER_GET', 'USER_PATCH', 'USER_POST'];
```
Comment (henryzarza) [inline react/react-training/src/redux/user/actions.js]: The same here, this variable isn't necessary

File: react/react-training/src/redux/user/actions.js [hunk 7]
```diff
@@ -1,109 +1,28 @@
...
-        type: actionTypes.USER_GET_FAILURE,
-        payload: {
-          userDataError
-        }
-      });
-    }
-  },
+  userGet: userId => ({
+    type: actionTypes.USER_GET,
+    target: 'userData',
```
Comment (henryzarza) [inline react/react-training/src/redux/user/actions.js]: Move this to constant variable

File: react/react-training/src/redux/user/actions.js [hunk 8]
```diff
@@ -1,109 +1,28 @@
...
-        type: actionTypes.USER_PATCH_FAILURE,
-        payload: {
-          userDataError
-        }
-      });
-    }
-  },
+  userPatch: (userId, newUserData) => ({
+    type: actionTypes.USER_PATCH,
+    target: 'userData',
```
Comment (henryzarza) [inline react/react-training/src/redux/user/actions.js]: The same here

File: react/react-training/src/redux/user/actions.js [hunk 9]
```diff
@@ -1,109 +1,28 @@
...
-        type: actionTypes.USER_POST_FAILURE,
-        payload: {
-          userDataError
-        }
-      });
-    }
-  }
+  userPost: myUser => ({
+    type: actionTypes.USER_POST,
+    target: 'userData',
```
Comment (henryzarza) [inline react/react-training/src/redux/user/actions.js]: And here

**Review by henryzarza — CHANGES_REQUESTED** [review]
Excellent! congratulations only a few comments and we're ready.

File: react/react-training/src/constants/index.js [hunk 10]
```diff
@@ -6,6 +6,12 @@ export const LINES = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2,
 
 export const OFFLINE_MESSAGE = 'Sorry. We are offline. Please come back in a few minutes.';
 
+export const TARGE_USERDATA = 'userData';
+
+export const TARGE_USERISLOGGED = 'userIsLogged';
+
+export const TARGET_USERLOGIN = 'userLogin';
```
Comment (wfolini) [inline react/react-training/src/constants/index.js]: it isn't necessary to declare this const here if you aren't going to use it in many files. This constants are used only in one file. So, I'd move this three constants to that files

**Review by wfolini — CHANGES_REQUESTED** [review]
Great! Just one comment

**Review by wfolini — APPROVED** [review]
👌

**Review by henryzarza — APPROVED** [review]

**Pull request merged**